include/mxnet/c_api.h

```diff
@@ -1423,13 +1423,15 @@ MXNET_DLL int MXSymbolInferType(SymbolHandle sym,
  * \param excluded_symbols array of symbols to be excluded from being quantized
  * \param num_offline number of parameters that are quantized offline
  * \param offline_params array of c strings representing the names of params quantized offline
+ * \param dev_type device type 
  */
 MXNET_DLL int MXQuantizeSymbol(SymbolHandle sym_handle,
                                SymbolHandle *ret_sym_handle,
                                const mx_uint num_excluded_symbols,
                                const SymbolHandle *excluded_symbols,
                                const mx_uint num_offline,
-                               const char **offline_params);
+                               const char **offline_params,
+                               int dev_type);
```

Comment:
Is this used to differentiate signed or unsigned quantization? If so, does it make more sense to define this as a boolean variable indicating whether it's signed or unsigned? After all, GPU could have unsigned quantization as well.

Reply:
Please also set default value for the variable for backward compatibility.

Reply:
@reminisce Currently the difference is mainly signed vs unsigned, but there maybe more difference in the future between CPU and GPU (like we may don't need requantize for CPU context), so looks we should use QuantizeGraphCPU as the name instead of QuantizeGraphUnsigned.  And the default value has set to cpu context in _quantize_symbol().

Reply:
@reminisce 
I understand that for other kind of CPU it may need requantize, how about below approach:
(1) add a few options for ```imagenet_gen_qsym.py``` to configure some features, like ```use_uint8``` (default off, to indicate whether quantize data to uint8), ```use_requantize``` (default on, to indicate whether we need requantize OP, to be added later), ```calib_input_data``` (default off, this is a feature we plan to add to enable input data calibration for calib layers to improve performance, to be added later), so users can pass different options to the quantization script based on their need and in most cases the default values will just work fine 
(2) instead of passing dev_type in ```MXQuantizeSymbol()```, we can add parameters to pass specific feature/function on/off configuration, like ```use_uint8```, ```use_requantize```, ```calib_input_data``` which mapped to quantization script options in (1)
(3) in C++ handling function ```MXQuantizeSymbol()```, define different graph attributes for above feature/function configurations, and in ```QuantizeGraph()``` perform quantize graph logic based on graph attribute values, this could make code reuse so we don't need to define seperate quantize graph function for CPU path.

Please let me know your comments/suggestions, thanks.

Reply:
@jinhuang415 I think this design is clearer than before. I agree with adding config params to the interface. One minor suggestion, instead of defining boolean value `use_uint8`, it's more general to define a param such as `quantized_dtype` default to `int8`. This allows us to pass other low-precision data types to the backend.